public/app/features/plugins/extensions/registry/ExposedComponentsRegistry.ts

```diff
@@ -0,0 +1,72 @@
+import { PluginExposedComponentConfig } from '@grafana/data';
+
+import { PluginPreloadResult } from '../../pluginPreloader';
+import { logWarning } from '../utils';
+
+import { Registry } from './Registry';
+
+export type RegistryType = {
+  [id: string]: {
+    pluginId: string;
+    config: PluginExposedComponentConfig;
+  };
+};
+
+export class ExposedComponentsRegistry extends Registry<RegistryType> {
+  constructor(initialState: RegistryType = {}) {
+    super({
+      initialState,
+    });
+  }
+
+  mapToRegistry(registry: RegistryType, item: PluginPreloadResult): RegistryType {
+    const { pluginId, exposedComponentConfigs, error } = item;
+
+    if (error) {
+      logWarning(`"${pluginId}" plugin failed to load. Skip registering its exposed components.`);
+      return registry;
+    }
```

Comment:
I was thinking about handling these kind of errors in the `pluginPreloader` instead of pushing them down here (I think it belongs more there as it's plugin-loading related, and would simplify this part of the code).
What do you think?

Reply:
Resolving this as probably [this diff](https://github.com/grafana/grafana/compare/extensions/refactor-exported-components...leventebalogh/extensions/refactor-exported-components) mentioned in the main comment explains what I mean more. (Reopen the thread if you don't think so.)

Reply:
Totally agree 👍